### PR TITLE
[tcgc] support value type for client default value

### PR DESCRIPTION
**Pull request by tadelesh**
fix: https://github.com/Azure/typespec-azure/issues/1598

### CODE REVIEW

File: packages/typespec-client-generator-core/src/internal-utils.ts [hunk 1]
```diff
@@ -569,3 +569,30 @@ export function isOnClient(
...
+      return value.value.value ?? value.value.name;
+    case "ObjectValue":
+      return Object.fromEntries(
+        [...value.properties.keys()].map((x) => [
+          x,
+          getValueTypeValue(value.properties.get(x)!.value),
+        ]),
+      );
+    case "ScalarValue":
+      // TODO: handle scalar value
```
Comment (haolingdong-msft) [inline packages/typespec-client-generator-core/src/internal-utils.ts:595]: handle this TODO?

**Review by iscai-msft — CHANGES_REQUESTED** [review]
the `value` in tsp purposefully is meant to describe service default values, not client default values, so we shouldn't be having this support. There are a lot of issues with roundtrippability with default values, so we should only have `clientDefaultValue` in the specific cases of endpoint values and client api versions. Where is the push for this support coming from?

**Comment by tadelesh** [comment]
> the `value` in tsp purposefully is meant to describe service default values, not client default values, so we shouldn't be having this support. There are a lot of issues with roundtrippability with default values, so we should only have `clientDefaultValue` in the specific cases of endpoint values and client api versions. Where is the push for this support coming from?

see the issue linked in description. there are cases that endpoint default value use value types, the support is only for those cases.

**Comment by iscai-msft** [comment]
ah ok, sorry for the confusion, let me remove the blocking pr

**Review by iscai-msft — APPROVED** [review]

**Pull request merged**